tests/python/relay/test_tidl.py

```diff
@@ -0,0 +1,210 @@
+from matplotlib import pyplot as plt
+import tensorflow as tf
+from tvm.relay.testing import tf as tf_testing
+from gluoncv import model_zoo, data, utils
+from gluoncv.data.transforms.presets.segmentation import test_transform
+from tvm.relay.backend.contrib import tidl
+
+def model_compile(model_name, mod_orig, params, 
+                  data_layout, input_node, input_data):
+    artifacts_folder = "./artifacts_" + model_name
+    if os.path.isdir(artifacts_folder):
+        filelist = [ f for f in os.listdir(artifacts_folder)]
+        for file in filelist:
+            os.remove(os.path.join(artifacts_folder, file))
+    else:
+        os.mkdir(artifacts_folder)
+
+    if force_arm_only is True:
+        mod = mod_orig
+        tidlEnabled = False
+    else:
+        tidl_compiler = tidl.TIDLCompiler("AM57", (6,3), 
+                                          num_tidl_subgraphs = num_tidl_subgraphs, 
+                                          data_layout = data_layout, 
+                                          artifacts_folder = artifacts_folder) 
+        input = {input_node: input_data}
+        mod = tidl_compiler.enable(mod_orig, params, input)
+        tidlEnabled = True
+        if mod == None:  # TIDL cannot be enabled - no offload to TIDL
+            mod = mod_orig
+            tidlEnabled = False
+
+    graph, lib, params = relay.build_module.build(mod, target=target, params=params)
+    path_lib    = os.path.join(artifacts_folder, "deploy_lib.so")
+    path_graph  = os.path.join(artifacts_folder, "deploy_graph.json")
+    path_params = os.path.join(artifacts_folder, "deploy_param.params")
+    lib.export_library(path_lib, cc=arm_gcc)
+    with open(path_graph, "w") as fo:
+      fo.write(graph)
+    with open(path_params, "wb") as fo:
+      fo.write(relay.save_param_dict(params))
+
+    return tidlEnabled
+
+def load_gluoncv_model(model, x, input_name, input_shape, dtype):
+    block = model_zoo.get_model(model, pretrained=True)
+
+    if 'faster' in model:
+        mod, params = relay.frontend.from_mxnet(block, shape={input_name: input_shape}, dtype=dtype)
+    else:
+        block.hybridize()
+        block.forward(x)
+        block.export('temp') # create file temp-symbol.json and temp-0000.params
+
+        model_json = mx.symbol.load('temp-symbol.json')
+        save_dict = mx.ndarray.load('temp-0000.params')
+        arg_params = {}
+        aux_params = {}
+        for k, v in save_dict.items():
+            tp, name = k.split(':', 1)
+            if tp == 'arg':
+                arg_params[name] = v
+            elif tp == 'aux':
+                aux_params[name] = v
+        mod, params = relay.frontend.from_mxnet(model_json, {input_name: input_shape}, arg_params=arg_params, aux_params=aux_params)
+    return block, mod, params
+
+def test_tidl_classification(model_name):
+    dtype = 'float32'
+    input_shape = (1, 3, 224, 224)
+    data_layout = "NCHW"
+    input_node  = "data"
+    input_image = "./dog.npy"

```

Comment:
Can you use `download_testdata` to download an image (can be street_small.jpg)?

Reply:
Done